### PR TITLE
Add missing dependencies

### PR DESCRIPTION
**Pull request by dmentipl**
Add missing dependencies.

Addresses #1.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by dmentipl** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -20,12 +20,19 @@ requirements:
...
     - cftime ==1.0.1
+    - cf_units
     - clize
+    - dask
     - iris >=2.2
+    - netCDF4
     - numpy
+    - python-dateutil
     - python-stratify
+    - pytz ==2020.5
```
Comment (lucyleeow) [inline recipe/meta.yaml:33]: Similar question about having pytz be optional as it is only used for ancillary files

**Review by lucyleeow — APPROVED** [review]
LGTM, thanks!

File: recipe/meta.yaml (hunk 1 above)
Comment (dmentipl) [reply recipe/meta.yaml:33]: See here: https://github.com/metoppv/improver/pull/1436#discussion_r603009495

Comment (lucyleeow) [reply recipe/meta.yaml:33]: Thanks, makes sense. I did not realise timezonefinder (also only used for generating ancillary files) is only imported in a function.

**Review by tjtg — APPROVED** [review]
All looks good.

**Comment by dmentipl** [comment]
I have now pinned pytz as unpinning it was causing the acceptance tests to fail: see here https://github.com/metoppv/improver/pull/1436#pullrequestreview-622942207.

**Review by tjtg — APPROVED** [review]

**Pull request merged**